### DISCUSSION TITLE
FR: Extend Syslog messages

### DISCUSSION THREAD

**Discussion by stevenengland** [arendst/Tasmota#19571] [Ideas]
Thanks for the great project!

### PROBLEM DESCRIPTION
Syslog messages do not provide facility, facility_num, level
See screenshot of a captured message sent to graylog

![image](https://github.com/arendst/Tasmota/assets/11787949/60f880b0-c02d-43de-8b2a-ed94637a1e44)

Would be great to have those fields (at least level) in place. So that one can easily distinguish messages that are of level 4 or higher and more...

**Comment by Frogmore42** [comment]
I just tried syslog (to my Synology NAS) and noticed that level did not seem right.  Looking at the Wikipedia page, facility is determined based on facility_num.  It is up to the syslog server to do the translation from number to name.  Tasmota does set the facility number (it is hard-coded to 16, which is local0).  That leaves level.

The function ```GetLog()``` in support.ino does have access to the log level and could pass it back to its callers.  This would be a breaking change, since the function parameters would need to change to do this, or it could be a new function (```GetLogAndLevel()```?) with ```GetLog()``` just calling it.

The levels in Tasmota:

- More Debug - 4
- Debug - 3
- Info - 2
- Error - 1
- None - 0

For syslog
- debug - 7
- info - 6
- notice - 5
- warning - 4
- error - 3
- critical - 2
- alert - 1
- emerg - 0

This easy translation seems to make sense to me.  Tasmota loglevel + 3 ==> syslog level:
More Debug ==> debug
Debug ==> info
Info ==> notice
Error ==> warning

I think it is important to make more debug different from debug, but maybe not.  If not, this could work:
More Debug ==> debug
Debug ==> debug
Info ==> info
Error ==> error

This could also work:
More Debug ==> debug
Debug ==> info
Info ==> notice
Error ==> error

Any thoughts/preferences?
